ixwebsocket/IXWebSocketMessageQueue.h

```diff
@@ -0,0 +1,52 @@
+/*
+ *  IXWebSocketMessageQueue.h
+ *  Author: Korchynskyi Dmytro
+ *  Copyright (c) 2017-2019 Machine Zone, Inc. All rights reserved.
+ */
+
+#pragma once
+
+#include "IXWebSocket.h"
+#include <thread>
+#include <list>
+#include <memory>
+
+namespace ix
+{
+    //
+    // A helper class to dispatch websocket message callbacks in your thread.
+    //
+    class WebSocketMessageQueue
+    {
+    public:
+        WebSocketMessageQueue(WebSocket* websocket = nullptr);
+        ~WebSocketMessageQueue();
+
+        void bindWebsocket(WebSocket* websocket);
+
+        void setOnMessageCallback(const OnMessageCallback& callback);
+
+        void poll(int count = 512);
+
+    protected:
+        struct Message
```

Comment:
@bsergean what about using the Message struct in IXW and pass it as shared_ptr?
This will eliminate redundant copy of data which might be long.

Reply:
Yes I was thinking about that. This would simplify all APIs to only take a 'BigMessage'. We'd have to put that struct or class into its own header file.

Reply:
BTW we should do that in a different PR, because this will be a very messy change.

Reply:
Totally agree

Reply:
Also we should think about whether this will create more allocs when copying those small Messages (mostly for the str).

Reply:
string in any case allocates memory

Reply:
for short string might not, this depends on string implementation